cvat-core/src/frames.ts

```diff
@@ -65,9 +71,30 @@ export class FramesMetaData {
             stop_frame: undefined,
         };
 
+        const updateTrigger = new FieldUpdateTrigger();
+        this.#updateTrigger = updateTrigger;
+
         for (const property in data) {
             if (Object.prototype.hasOwnProperty.call(data, property) && property in initialData) {
-                data[property] = initialData[property];
+                if (property === 'deleted_frames') {
+                    const handler = {
+                        set(target, prop, value) {
+                            target[prop] = value;
+                            updateTrigger.update('deletedFrames');
+                            return true;
+                        },
+                        deleteProperty(target, prop) {
+                            const result = delete target[prop];
+                            if (result) {
```

Comment:
Shall we additionally check if the property exists?  It would be safer, because in general JS allows deleting not-existing properties. 

Also, this check below:

```
if (frame in meta.deletedFrames) {
        delete meta.deletedFrames[frame];
    }
```

may be removed, leaving just `delete` operator